src/persist-client/src/internal/state.rs

```diff
@@ -472,18 +503,48 @@ impl<T> HollowBatch<T> {
         desc: Description<T>,
         parts: Vec<BatchPart<T>>,
         len: usize,
-        runs: Vec<usize>,
+        run_meta: Vec<RunMeta>,
+        run_splits: Vec<usize>,
```

Comment:
You could combine these by changing the definition of `run_splits`; preserving the lengths or the end indices instead of just the points where two runs meet. More invasive but probably less janky; up to you.

Reply:
Would be cleaner, but doesn't seem like a big enough win to risk potentially introducing a bug in the refactor